docs/tracing.md

```diff
@@ -0,0 +1,306 @@
+|                                           |
+| +-----------+                             |
+| | Trace     |                             |
+| | Collector |                             |
+| +-----+-----+                             |
+|       ^                  +--------------+ |
+|       | spans            | Kata VM      | |
+| +-----+-----+            |              | |
+| | Kata      |    spans   |     +-----+  | |
+| | Trace     |<-----------------|Kata |  | |
+| | Forwarder |    VSOCK   |     |Agent|  | |
+| +-----------+    Channel |     +-----+  | |
+|                          +--------------+ |
++-------------------------------------------+
+```
+
+# Agent tracing terminology
+
+## Trace modes
+
+The agent supports two different tracing modes:
+
+| Trace mode | Description | Use-case | Setting | VM shutdown controller | Notes |
+|-|-|-|-|-|-|
+| Static | Traces from agent start to agent shutdown. | Holistic view of agent activities. Possibly more suited to short-running containers. | `trace_mode = "static"` | agent | Default trace mode. |
+| Dynamic | Traces begin when the ttRPC `StartTracing()` API is called and end when the corresponding `StopTracing()` API is called. | On-demand (partial) tracing. More appropriate for long-running containers. | `trace_mode = "dynamic"` | runtime | Only "isolated" type supported. |
+
+## Trace types
+
+Each trace mode is sub-divided into two different types:
+
+| Trace type | Description | Use-case | Setting | Notes |
+|-|-|-|-|-|
+| isolated | The traces only apply to the agent: the first span is generated at agent startup and the last at agent shutdown. | Observing agent lifespan. | `trace_type = "isolated"` | Default trace type. |
+| collated | In this mode, spans are associated with their Kata runtime initiated counterparts. | Understanding how the runtime and agent interact. | `trace_type = "collated"` | Also requires runtime tracing to be enabled in `configuration.toml` (`enable_tracing=true`). |
+
+# Prerequisites
+
+- You must have a trace collector running on the host.
+
+  The quickest and simplest method for doing this is to use the
+  [Jaeger "all-in-one" Docker image][jaeger-all-in-one].
+
+- If you wish to trace the agent, you must start the
+  [trace forwarder][trace-forwarder].
+
+> **Notes:**
+>
+> - If agent tracing is enabled but the forwarder is not running,
+>   the agent will exit (panic) with an error message. Since the agent is not
+>   directly visible by the user, it is recommended to enable debug to capture
+>   this message in the logs.
+>
+> - The forwarder requires a trace collector (such as Jaeger) to be running
+>   before it is started. If a collector is not running, the trace forwarder
+>   will exit with an error.
+
+# Enable tracing
+
+By default, tracing is disabled for all components. To enable _any_ form of
+tracing the `enable_tracing` option must be enabled.
+
+> **Note:** 
+>
+> Enabling this option will only allow tracing for subsequently
```

Comment:
Is this a problem in the shimv2 case, if I had another pod running before I decided to enable tracing?